CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,559 @@
+      unsigned nsh = ch->nStrawHits();
+      tc._nsh += nsh;
+      const XYZVectorF& pos = ch->pos();
+      float hwt = ch->nStrawHits();
+      tmin(htime);
+      tmax(htime);
+      tacc(htime,weight=hwt);
+      xacc(pos.x(),weight=hwt);
+      yacc(pos.y(),weight=hwt);
+      zacc(pos.z(),weight=hwt);
+    }
+
+    static float invsqrt12(1.0/sqrt(12.0));
+    tc._t0._t0 = extract_result<tag::weighted_median>(tacc);
+    tc._t0._t0err = ( boost::accumulators::extract::max(tmax)-boost::accumulators::extract::min(tmin))*invsqrt12/sqrt(nstrs);
+    tc._pos = XYZVectorF(extract_result<tag::weighted_median>(xacc),
+    extract_result<tag::weighted_median>(yacc),
+    extract_result<tag::weighted_median>(zacc));
+    // if (_debug > 2) std::cout<<"Init Cluster T0 = "<<tc._t0._t0<<" n straw hits = "<<tc._nsh<<" n combo hits = "<<tc._strawHitIdxs.size()<<std::endl;
+  }
+
+//------------------------------------------------------------------------------
+// Function to find the sigma of phi and time clusters. Note : Does not work yet for the clusters where some hits are close to 0 and some close to 2pi
+//-----------------------------------------------------------------------------
+  float PhiClusterFinder::checkdelta(TimeCluster& tc,int ClustNo, const std::vector<StrawHitIndex>& ordchcol) {
+    float meanphi(0),sigphi(0);//,sig(0);
+    for(auto ish : tc._strawHitIdxs) {
+      const ComboHit* ch = &_chcol->at(ish);
+      float phi = ch->phi();
+      if(phi < 0) phi += 2*M_PI;
+      meanphi = meanphi + phi;
+    }
+    meanphi = meanphi/tc._strawHitIdxs.size();
+    for(auto ish :tc._strawHitIdxs) {
+      const ComboHit* ch = &_chcol->at(ish);
+      float phi = ch->phi();
+      if(phi < 0) phi += 2*M_PI;
+      sigphi = sigphi + fabs(meanphi-phi);
+    }
+    sigphi = sigphi/(tc._strawHitIdxs.size()-1);
+    // if(_debug>3) std::cout<<"Check Delta time sigma = "<<sig<<" phi sigma = "<<sigphi<<" t0 = "<<tc._t0._t0<<" n straw hits = "<<tc._nsh<<" n combo hits = "<<tc._strawHitIdxs.size()<<std::endl;
+    return sigphi;
+  }
+
+//------------------------------------------------------------------------------
+// Function to add calo cluster to the time clusters
+//-----------------------------------------------------------------------------
+  void PhiClusterFinder::addCaloClusters(TimeClusterCollection& tccolnew, const TimeClusterCollection& tccol) {
+    for(size_t ipeak=0; ipeak<tccol.size(); ipeak++) {
+      auto& tc = tccol[ipeak];
+      if (tc.hasCaloCluster()) {
+       // if(_debug>1) std::cout<<"TC calo x = "<<tc._caloCluster->cog3Vector().x()<<" y = "<<tc._caloCluster->cog3Vector().y()<<" time = "<<tc._caloCluster->time()<<" nhits = "<<tc.nhits()<<std::endl;
+        for(size_t inewpeak = 0; inewpeak<tccolnew.size();inewpeak++){
+         auto& newtc = tccolnew[inewpeak];
+         float dphi = fabs(polyAtan2( newtc._pos.y(), newtc._pos.x()) - polyAtan2( tc._caloCluster->cog3Vector().y(), tc._caloCluster->cog3Vector().x()));
```

Comment:
the evaluation of `polyAtan2( tc._caloCluster->cog3Vector().y(), tc._caloCluster->cog3Vector().x())` can be performed outside of the inner `for` loop to avoid re-doing it multiple times.